### PR TITLE
Global navigation

### LINKED ISSUE #491: GlobalNavLink: Test that URL/href gets passed to callback as argument
Link `href` has been moved from event property to additional callback parameter.

context:
https://github.com/salesforce-ux/design-system-react/pull/484/files#diff-6c2a578959daeaeb1c9534a7d7a071feR221

### LINKED ISSUE #1647: Global Navigation: Align markup and UX pattern to SLDS
This issue replaces #579. #579 is so out of date that I don't it to be the primary issue, but it should be reviewed.

Global Navigation has not been updated in two years. It should be updated to current SLDS UX patterns and flexibly meet the needs of any product planning to use it.

https://www.lightningdesignsystem.com/components/global-navigation/
* Align to current SLDS markup and design
* Align to current SLDS UX patterns (default answer is look to platform)
* Review Salesforce platform navigation use to be aware of common UX patterns.
* Fix all accessibility issues and move to production component

For right now, tab feature can be excluded from scope.

## Errata
* Ramp up, no tests, get UX patterns correct for demo work
* Primarily markup alignment (button and dropdowns)
* Probably doesn't need new APIs
* Review platform for consistency

### PR DESCRIPTION
**Pull request by kevinparkerson**
Fixes #1647

Also Fixes #491 by testing second `data` argument in callback for `href` string attribute

Updates to markup and tests. Some notable changes:
- removed "slds-context-bar-action__trigger" and iconSize="x-small" from global-navigation-bar/dropdown-trigger button
- removed "slds-text-title_caps" from menu-headers in utilities/menu-list/item
- added title={props.title} to utilities/menu-list/item-label

NOTE: while the expected markup is now correct, visually the global nav does not look like current SLDS in all areas due to DSR being on an older version of SLDS. When PR #1750 is merged these visual discrepancies will disappear.

---

### CONTRIBUTOR checklist (do not remove)
Please complete for every pull request

* [x] First-time contributors should sign the Contributor License Agreement. It's a fancy way of saying that you are giving away your contribution to this project. If you haven't before, wait a few minutes and a bot will comment on this pull request with instructions.
* [x] `npm run lint:fix` has been run and linting passes.
* [x] Mocha, Jest (Storyshots), and `components/component-docs.json` CI tests pass (`npm test`).
* [x] Tests have been added for new props to prevent regressions in the future. See [readme](https://github.com/salesforce/design-system-react/blob/master/tests/README.md).
* [x] Review the appropriate Storybook stories. Open [http://localhost:9001/](http://localhost:9001/).
* [x] The Accessibility panel of each Storybook story has 0 violations (aXe). Open [http://localhost:9001/](http://localhost:9001/).
* [x] Review tests are passing in the browser. Open [http://localhost:8001/](http://localhost:8001/).
* [x] Review markup conforms to [SLDS](https://www.lightningdesignsystem.com/) by looking at [DOM snapshot strings](https://facebook.github.io/jest/docs/en/snapshot-testing.html).

### REVIEWER checklist (do not remove)

* [ ] TravisCI tests pass. This includes linting, Mocha, Jest, Storyshots, and `components/component-docs.json` tests.
* [ ] Tests have been added for new props to prevent regressions in the future. See [readme](https://github.com/salesforce/design-system-react/blob/master/tests/README.md).
* [ ] Review the appropriate Storybook stories. Open [http://localhost:9001/](http://localhost:9001/).
* [ ] The Accessibility panel of each Storybook story has 0 violations (aXe). Open [http://localhost:9001/](http://localhost:9001/).
* [ ] Review tests are passing in the browser. Open [http://localhost:8001/](http://localhost:8001/).
* [ ] Review markup conforms to [SLDS](https://www.lightningdesignsystem.com/) by looking at [DOM snapshot strings](https://facebook.github.io/jest/docs/en/snapshot-testing.html).
###### Required only if there are markup / UX changes
* [ ] Add year-first date and commit SHA to `last-slds-markup-review` in `package.json` and push.
* [ ] Request a review of the deployed Heroku app by the Salesforce UX Accessibility Team.
* [ ] Add year-first review date, and commit SHA, `last-accessibility-review`, to `package.json` and push.
* [ ] While the contributor's branch is checked out, run `npm run local-update` within locally cloned [site repo](https://github.com/salesforce-ux/design-system-react-site) to confirm the site will function correctly at the next release.

### CODE REVIEW

**Comment by futuremint** [comment]
@kevinparkerson have you had accessibility look at this?  Using VoiceOver I can't read the name of the app... it focuses on the App Launcher waffle, over the app name and right into secondary navigation.

File: components/global-navigation-bar/dropdown-trigger.jsx [hunk 1]
```diff
@@ -179,17 +179,20 @@ class GlobalNavigationDropdownTrigger extends React.Component {
 				style={listItemstyle}
 			>
 				{/* eslint-enable jsx-a11y/no-static-element-interactions */}
```
Comment (futuremint) [inline components/global-navigation-bar/dropdown-trigger.jsx]: This is disabled above, and then re-enabled here.  Can we try to eliminate both of these while we're in this code and updating it?

File: components/global-navigation-bar/index.jsx [hunk 2]
```diff
@@ -96,10 +84,7 @@ GlobalNavigationBar.propTypes = {
 	theme: PropTypes.oneOf(['light', 'dark']),
 };
 
-GlobalNavigationBar.defaultProps = {
-	cloud: 'default',
-	theme: 'dark',
-};
```
Comment (futuremint) [inline components/global-navigation-bar/index.jsx]: Why did you remove these?  These are convenient if a developer has to write less code.

**Comment by kevinparkerson** [comment]
> We need to add to the stories a menu item with an icon in it so that we can cover section [1.3 from the SLDS site](https://lightningdesignsystem.com/components/global-navigation/)

@futuremint wouldn't the base example satisfy that?
screen shot 2019-03-07 at 2 59 11 pm

**Comment by kevinparkerson** [comment]
> @kevinparkerson have you had accessibility look at this? Using VoiceOver I can't read the name of the app... it focuses on the App Launcher waffle, over the app name and right into secondary navigation.

@futuremint not yet but I'll talk with JoAnne and see if we can get a review done before next week!

File: components/global-navigation-bar/index.jsx (hunk 2 above)
Comment (kevinparkerson) [reply components/global-navigation-bar/index.jsx]: For `cloud`, the default appearance is assumed if nothing is set. The SLDS class `.slds-context-bar_theme-default` does not exist, but `.slds-context-bar_theme-marketing` does. Right now the `cloud` PropType is `string`, but perhaps it should be more specific such as `cloud: PropTypes.oneOf(['marketing'])`?

For `theme`, I'm actually not sure if this is still needed after re-review. At the very least I know the default is to not have any theme class applied, which is why I removed it here... but looking through the SLDS classes, I don't see any `.slds-context-bar_theme-light` or `.slds-context-bar_theme-dark` definitions... on the documentation site there's only one reference to those here:
screen shot 2019-03-07 at 3 53 17 pm

Perhaps we should get rid of theme altogether? @interactivellama, do you happen to know anything else about the light / dark themes?

File: components/global-navigation-bar/dropdown-trigger.jsx (hunk 1 above)
Comment (kevinparkerson) [reply components/global-navigation-bar/dropdown-trigger.jsx]: Turns out those weren't needed in the first place! Removed them and ran into 0 linting errors without any additional work.

File: components/global-navigation-bar/index.jsx (hunk 2 above)
Comment (interactivellama) [reply components/global-navigation-bar/index.jsx]: I'm not seeing any reference to light and dark themes in the CSS either.

**Comment by interactivellama** [comment]
I think the best way to have VoiceOver read the App name since it's not a link is to go down into the grouping with Ctrl Shift Opt Down and then control Opt right.
